node/pkg/proto/gossip/v1/gossip.pb_test.go

```diff
@@ -0,0 +1,119 @@
+package gossipv1
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/hex"
+	vaa "github.com/certusone/wormhole/node/pkg/vaa"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"testing"
+
+	"time"
+)
+
+func getVAA() vaa.VAA {
+	var payload = []byte{97, 97, 97, 97, 97, 97}
+	var governanceEmitter = vaa.Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+
+	vaa := vaa.VAA{
+		Version:          uint8(1),
+		GuardianSetIndex: uint32(1),
+		Signatures:       nil,
+		Timestamp:        time.Unix(0, 0),
+		Nonce:            uint32(1),
+		Sequence:         uint64(1),
+		ConsistencyLevel: uint8(32),
+		EmitterChain:     vaa.ChainIDSolana,
+		EmitterAddress:   governanceEmitter,
+		Payload:          payload,
+	}
+
+	return vaa
+}
+
+func TestSignedObservation(t *testing.T) {
+	vaa := getVAA()
+	digest := vaa.SigningMsg()
+
+	txhash := []byte{97, 97, 97, 97, 97, 97}
+	signature := []byte{97, 97, 97, 97, 97, 97}
+
+	privKey, _ := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+
+	obsv := SignedObservation{
+		Addr:      crypto.PubkeyToAddress(privKey.PublicKey).Bytes(),
+		Hash:      digest.Bytes(),
+		Signature: signature,
+		TxHash:    txhash,
+		MessageId: vaa.MessageID(),
+	}
+
+	assert.Equal(t, signature, obsv.Signature)
+	assert.Equal(t, txhash, obsv.TxHash)
+	assert.Equal(t, vaa.MessageID(), obsv.MessageId)
```

Comment:
You assert that `SignedObservation.Signature` was set to the signature which you set. You then assert that `SignedObservation.TxHash` was set to the txhash you set. Lastly you assert that `vaa.MessageID()` is equal to the obvs.MessageId that you set to `vaa.MessageID()`.

This test is testing that go sets structs to what you set them to. There is no beneficial value in this. Consider removing this test entirely. Testing that setting structs works should be in go's stdlib tests, not in application tests. Consider not doing this.